### PR TITLE
Feat(web): Introduce Icon component #DS-286

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

**Comment by literat** [comment]
@crishpeen Please, run `yarn format` to reformat Markdown.

**Review by literat — APPROVED** [review]

File: packages/web/src/scss/components/Icon/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Icon
+
+Use Icon component to display an icon from our icon set. This icon set is
+exported from Supernova into separate `spirit-icons` package. So make sure you
+install it first and then copy the icons into your project.
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Icon/README.md:5]: > So make sure you install it first and then copy the icons into your project.

That should not be necessary, `spirit-icons` are in the dependencies of this project. Or do I get it wrong?

File: packages/web/src/scss/components/Icon/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+You can use sprite with all the icons.
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Icon/README.md:7]: I'd either rephrase or remove this. Now it sounds like an option, but sprites are currently the _only_ way I can use icons.

File: packages/web/package.json [hunk 3]
```diff
@@ -15,11 +15,12 @@
   "main": "js/cjs/spirit-web.js",
   "module": "js/esm/spirit-web.js",
   "scripts": {
+    "prestart": "shx cp -r ../../node_modules/@lmc-eu/spirit-icons/dist/ dist/icons",
```
Comment (adamkudrna) [inline packages/web/package.json:18]: Would it make any sense to have here the same as in the `prebuild` script?

File: packages/web/src/scss/components/Icon/index.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{{#> layout }}
+  <h2 class="typography-heading-medium-text mt-1100 mb-800">Icon</h2>
+  <h3 class="typography-heading-small-text mt-1100 mb-800">Design only</h3>
+  <svg class="Icon" width="24" height="24">
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Icon/index.html:4]: 👀 
obrazek

**Review by adamkudrna — CHANGES_REQUESTED** [review]

File: packages/web/src/scss/components/Icon/README.md (hunk 1 above)
Comment (crishpeen) [reply packages/web/src/scss/components/Icon/README.md:5]: Removed

File: packages/web/package.json (hunk 3 above)
Comment (crishpeen) [reply packages/web/package.json:18]: Sure, I created new script and I start it from both pre* scripts

File: packages/web/src/scss/components/Icon/README.md (hunk 2 above)
Comment (crishpeen) [reply packages/web/src/scss/components/Icon/README.md:7]: Removed

File: packages/web/src/scss/components/Icon/index.html (hunk 4 above)
Comment (crishpeen) [reply packages/web/src/scss/components/Icon/index.html:4]: It works for me, but I sent an update. Can you try it again? Thanks. (btw `yarn (install)` is required and then `start` or `build`)

File: packages/web/package.json [hunk 5]
```diff
@@ -15,12 +15,16 @@
   "main": "js/cjs/spirit-web.js",
   "module": "js/esm/spirit-web.js",
   "scripts": {
+    "icons:build": "shx mkdir -p public/icons && shx cp -r ../../node_modules/@lmc-eu/spirit-icons/dist/. public/icons",
```
Comment (literat) [inline packages/web/package.json:18]:
```sh
6:06:42 PM: $ shx mkdir -p public/icons && shx cp -r ../../node_modules/@lmc-eu/spirit-icons/dist/. public/icons
6:06:43 PM: cp: no such file or directory: ../../node_modules/@lmc-eu/spirit-icons/dist/
```

On Netlify, the base directory is set to `packages/web` for this deploy, the `./node_modules` are in this base, I think.

File: packages/web/package.json [hunk 6]
```diff
@@ -15,12 +15,16 @@
...
+    "prestart": "shx mkdir -p public/icons && cd ../icons && yarn build && shx cp -r dist/. ../web/public/icons",
     "start": "vite --host",
+    "preexamples:build": "yarn icons:build",
     "examples:build": "vite build",
     "examples:build:gh": "yarn examples:build --base=/spirit-design-system/",
     "examples:preview": "vite preview",
     "prebuild": "shx rm -rf dist && shx mkdir -p dist/scss && shx cp package.json README.md dist/ && shx cp -r src/scss dist/",
     "build": "npm-run-all --serial css js",
+    "postbuild": "yarn icons:build",
```
Comment (literat) [inline packages/web/package.json:27]: Why do we need this `postbuild` and copying of icons? They are not needed in `./dist` and also not there.

**Review by literat — CHANGES_REQUESTED** [review]

File: packages/web/package.json (hunk 5 above)
Comment (crishpeen) [reply packages/web/package.json]: Fixed. I checked the node_modules folder in Netlify and it was basically empty. So I used the dist folder in icons.

**Review by adamkudrna — APPROVED** [review]

**Review by adamkudrna — COMMENTED** [review]
Ha! How about the size prop? I think we are going to have smaller icons in the product DS too.

**Comment by literat** [comment]
> Ha! How about the size prop? I think we are going to have smaller icons in the product DS too.

Just baby steps. When we will have those sizes we should implement the `size` prop, right?

File: packages/web/src/scss/components/Icon/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Icon
+
+Use Icon component to display an icon from our icon set. This icon set is
```
Comment (literat) [inline packages/web/src/scss/components/Icon/README.md:3]:
```suggestion
Use the Icon component to display an icon from our icon set. This icon set is
```

File: packages/web/package.json [hunk 8]
```diff
@@ -46,7 +49,8 @@
   },
   "dependencies": {
     "@csstools/normalize.css": "^12.0.0",
-    "@lmc-eu/spirit-design-tokens": "^0.19.0"
+    "@lmc-eu/spirit-design-tokens": "^0.19.0",
+    "@lmc-eu/spirit-icons": "^0.2.0"
```
Comment (literat) [inline packages/web/package.json:53]: I think we should add the same information to `packages/icons/README.md` as we have here https://github.com/lmc-eu/spirit-design-system/tree/main/packages/design-tokens#install. Could you do it, please?

**Comment by literat** [comment]
@crishpeen LGTM, great work ❤️ Just do a little bit of polishing 💅 as I suggested and it will be ✅ . Thanks.

**Review by literat — CHANGES_REQUESTED** [review]

**Comment by crishpeen** [comment]
> Ha! How about the size prop? I think we are going to have smaller icons in the product DS too.

Right now Spirit only has one icon size. So as @literat said, when we are going to need it, we will add it.

**Review by literat — APPROVED** [review]

**Pull request merged**